packages/grid/_modules_/grid/hooks/root/useGridApiEventHandler.ts

```diff
@@ -2,6 +2,25 @@ import * as React from 'react';
 import { MuiEvent } from '../../models/gridOptions';
 import { GridApiRef } from '../../models/api/gridApiRef';
 import { useLogger } from '../utils/useLogger';
+import { useGridSelector } from '../features/core/useGridSelector';
+import { optionsSelector } from '../utils/optionsSelector';
+import { GridApi } from '../../models/api/gridApi';
+
+// TODO: Remove once [[GridApi]] cycle dependency is fixed
+/**
+ * Callback details.
+ */
+export interface MuiCallbackDetails {
```

Comment:
This type is specific to the data grid, shouldn't it be?

```suggestion
export interface GridCallbackDetails {
```